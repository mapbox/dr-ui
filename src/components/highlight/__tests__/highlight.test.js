import React from 'react';
import renderer from 'react-test-renderer';
import { testCases } from './highlight-test-cases.js';

describe('syntax-highlighting', () => {
  describe(testCases.basic.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.basic;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.objc.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.objc;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.java.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.java;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });
});
