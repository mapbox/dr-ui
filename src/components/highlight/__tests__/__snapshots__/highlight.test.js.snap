// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`syntax-highlighting CSS highlighting renders as expected 1`] = `
<div
  data-swiftype-index="false"
>
  <div
    className="relative prose"
  >
    
    <div
      className="relative round z0 scroll-styled"
      style={Object {}}
    >
      <pre>
        <code
          className="px0 hljs"
        >
          <div
            className="relative z2"
            data-chunk-code="chunk-0"
          >
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 12,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<span class=\\"token selector\\">body</span> <span class=\\"token punctuation\\">{</span>",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 26.45,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<span class=\\"token property\\">margin</span><span class=\\"token punctuation\\">:</span> 0<span class=\\"token punctuation\\">;</span>",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 26.45,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<span class=\\"token property\\">padding</span><span class=\\"token punctuation\\">:</span> 0<span class=\\"token punctuation\\">;</span>",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 12,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<span class=\\"token punctuation\\">}</span>",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 12,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<span class=\\"token selector\\">#map</span> <span class=\\"token punctuation\\">{</span>",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 26.45,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<span class=\\"token property\\">position</span><span class=\\"token punctuation\\">:</span> absolute<span class=\\"token punctuation\\">;</span>",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 26.45,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<span class=\\"token property\\">top</span><span class=\\"token punctuation\\">:</span> 0<span class=\\"token punctuation\\">;</span>",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 26.45,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<span class=\\"token property\\">bottom</span><span class=\\"token punctuation\\">:</span> 0<span class=\\"token punctuation\\">;</span>",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 26.45,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<span class=\\"token property\\">width</span><span class=\\"token punctuation\\">:</span> 100%<span class=\\"token punctuation\\">;</span>",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 12,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<span class=\\"token punctuation\\">}</span>",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
          </div>
        </code>
      </pre>
      <div
        className="absolute z2 top right mr6 mt6 color-white"
      />
    </div>
  </div>
</div>
`;

exports[`syntax-highlighting Java highlighting renders as expected 1`] = `
<div
  data-swiftype-index="false"
>
  <div
    className="relative prose"
  >
    
    <div
      className="relative round z0 scroll-styled"
      style={Object {}}
    >
      <pre>
        <code
          className="px0 hljs"
        >
          <div
            className="relative z2"
            data-chunk-code="chunk-0"
          >
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 12,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<span class=\\"token keyword\\">public</span> <span class=\\"token keyword\\">class</span> <span class=\\"token class-name\\">MainActivity</span> <span class=\\"token keyword\\">extends</span> <span class=\\"token class-name\\">AppCompatActivity</span> <span class=\\"token punctuation\\">{</span>",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 26.45,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<span class=\\"token keyword\\">private</span> <span class=\\"token class-name\\">MapView</span> mapView<span class=\\"token punctuation\\">;</span>",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 12,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": " ",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 26.45,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<span class=\\"token annotation punctuation\\">@Override</span>",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 26.45,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<span class=\\"token keyword\\">protected</span> <span class=\\"token keyword\\">void</span> <span class=\\"token function\\">onCreate</span><span class=\\"token punctuation\\">(</span><span class=\\"token class-name\\">Bundle</span> savedInstanceState<span class=\\"token punctuation\\">)</span> <span class=\\"token punctuation\\">{</span>",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 26.45,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<span class=\\"token keyword\\">super</span><span class=\\"token punctuation\\">.</span><span class=\\"token function\\">onCreate</span><span class=\\"token punctuation\\">(</span>savedInstanceState<span class=\\"token punctuation\\">)</span><span class=\\"token punctuation\\">;</span>",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 26.45,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<span class=\\"token class-name\\">Mapbox</span><span class=\\"token punctuation\\">.</span><span class=\\"token function\\">getInstance</span><span class=\\"token punctuation\\">(</span><span class=\\"token keyword\\">this</span><span class=\\"token punctuation\\">,</span> <span class=\\"token string\\">\\"token!\\"</span><span class=\\"token punctuation\\">)</span><span class=\\"token punctuation\\">;</span>",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 26.45,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<span class=\\"token function\\">setContentView</span><span class=\\"token punctuation\\">(</span><span class=\\"token class-name\\">R</span><span class=\\"token punctuation\\">.</span>layout<span class=\\"token punctuation\\">.</span>activity_main<span class=\\"token punctuation\\">)</span><span class=\\"token punctuation\\">;</span>",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 26.45,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "mapView <span class=\\"token operator\\">=</span> <span class=\\"token punctuation\\">(</span><span class=\\"token class-name\\">MapView</span><span class=\\"token punctuation\\">)</span> <span class=\\"token function\\">findViewById</span><span class=\\"token punctuation\\">(</span><span class=\\"token class-name\\">R</span><span class=\\"token punctuation\\">.</span>id<span class=\\"token punctuation\\">.</span>mapView<span class=\\"token punctuation\\">)</span><span class=\\"token punctuation\\">;</span>",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 26.45,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "mapView<span class=\\"token punctuation\\">.</span><span class=\\"token function\\">onCreate</span><span class=\\"token punctuation\\">(</span>savedInstanceState<span class=\\"token punctuation\\">)</span><span class=\\"token punctuation\\">;</span>",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 12,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<span class=\\"token punctuation\\">}</span>",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
          </div>
        </code>
      </pre>
      <div
        className="absolute z2 top right mr6 mt6 color-white"
      />
    </div>
  </div>
</div>
`;

exports[`syntax-highlighting Objective C highlighting renders as expected 1`] = `
<div
  data-swiftype-index="false"
>
  <div
    className="relative prose"
  >
    
    <div
      className="relative round z0 scroll-styled"
      style={Object {}}
    >
      <pre>
        <code
          className="px0 hljs"
        >
          <div
            className="relative z2"
            data-chunk-code="chunk-0"
          >
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 12,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<span class=\\"token macro property\\"><span class=\\"token directive-hash\\">#</span><span class=\\"token directive keyword\\">import</span> <span class=\\"token string\\">\\"ViewController.h\\"</span></span>",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 12,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<span class=\\"token operator\\">@</span>import Mapbox<span class=\\"token punctuation\\">;</span>",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 12,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": " ",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 12,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<span class=\\"token keyword\\">@implementation</span> ViewController",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 12,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": " ",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 12,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<span class=\\"token operator\\">-</span> <span class=\\"token punctuation\\">(</span><span class=\\"token keyword\\">void</span><span class=\\"token punctuation\\">)</span>viewDidLoad <span class=\\"token punctuation\\">{</span>",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 26.45,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<span class=\\"token punctuation\\">[</span><span class=\\"token keyword\\">super</span> viewDidLoad<span class=\\"token punctuation\\">]</span><span class=\\"token punctuation\\">;</span>",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 12,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": " ",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 26.45,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "NSURL <span class=\\"token operator\\">*</span>url <span class=\\"token operator\\">=</span> <span class=\\"token punctuation\\">[</span>NSURL URLWithString<span class=\\"token punctuation\\">:</span><span class=\\"token operator\\">@</span>\\"mapbox<span class=\\"token punctuation\\">:</span><span class=\\"token comment\\">//styles/mapbox/streets-v10\\"];</span>",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 26.45,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "MGLMapView <span class=\\"token operator\\">*</span>mapView <span class=\\"token operator\\">=</span> <span class=\\"token punctuation\\">[</span><span class=\\"token punctuation\\">[</span>MGLMapView alloc<span class=\\"token punctuation\\">]</span> initWithFrame<span class=\\"token punctuation\\">:</span><span class=\\"token keyword\\">self</span><span class=\\"token punctuation\\">.</span>view<span class=\\"token punctuation\\">.</span>bounds styleURL<span class=\\"token punctuation\\">:</span>url<span class=\\"token punctuation\\">]</span><span class=\\"token punctuation\\">;</span>",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 26.45,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "mapView<span class=\\"token punctuation\\">.</span>autoresizingMask <span class=\\"token operator\\">=</span> UIViewAutoresizingFlexibleWidth <span class=\\"token operator\\">|</span> UIViewAutoresizingFlexibleHeight<span class=\\"token punctuation\\">;</span>",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 26.45,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<span class=\\"token punctuation\\">[</span>mapView setCenterCoordinate<span class=\\"token punctuation\\">:</span><span class=\\"token function\\">CLLocationCoordinate2DMake</span><span class=\\"token punctuation\\">(</span><span class=\\"token number\\">59.31</span><span class=\\"token punctuation\\">,</span> <span class=\\"token number\\">18.06</span><span class=\\"token punctuation\\">)</span>",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 26.45,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "zoomLevel<span class=\\"token punctuation\\">:</span><span class=\\"token number\\">9</span>",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 26.45,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "animated<span class=\\"token punctuation\\">:</span>NO<span class=\\"token punctuation\\">]</span><span class=\\"token punctuation\\">;</span>",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 26.45,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<span class=\\"token punctuation\\">[</span><span class=\\"token keyword\\">self</span><span class=\\"token punctuation\\">.</span>view addSubview<span class=\\"token punctuation\\">:</span>mapView<span class=\\"token punctuation\\">]</span><span class=\\"token punctuation\\">;</span>",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 12,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<span class=\\"token punctuation\\">}</span>",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 12,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": " ",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 12,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<span class=\\"token keyword\\">@end</span>",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
          </div>
        </code>
      </pre>
      <div
        className="absolute z2 top right mr6 mt6 color-white"
      />
    </div>
  </div>
</div>
`;

exports[`syntax-highlighting Swift highlighting renders as expected 1`] = `
<div
  data-swiftype-index="false"
>
  <div
    className="relative prose"
  >
    
    <div
      className="relative round z0 scroll-styled"
      style={Object {}}
    >
      <pre>
        <code
          className="px0 hljs"
        >
          <div
            className="relative z2"
            data-chunk-code="chunk-0"
          >
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 12,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<span class=\\"token keyword\\">import</span> <span class=\\"token builtin\\">Mapbox</span>",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 55.349999999999994,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<span class=\\"token keyword\\">class</span> <span class=\\"token class-name\\">ViewController</span><span class=\\"token punctuation\\">:</span> <span class=\\"token builtin\\">UIViewController</span> <span class=\\"token punctuation\\">{</span>",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 55.349999999999994,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<span class=\\"token keyword\\">override</span> <span class=\\"token keyword\\">func</span> <span class=\\"token function\\">viewDidLoad</span><span class=\\"token punctuation\\">(</span><span class=\\"token punctuation\\">)</span> <span class=\\"token punctuation\\">{</span>",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 55.349999999999994,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<span class=\\"token keyword\\">super</span><span class=\\"token punctuation\\">.</span><span class=\\"token function\\">viewDidLoad</span><span class=\\"token punctuation\\">(</span><span class=\\"token punctuation\\">)</span>",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 12,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": " ",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 55.349999999999994,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<span class=\\"token keyword\\">let</span> url <span class=\\"token operator\\">=</span> <span class=\\"token function\\">URL</span><span class=\\"token punctuation\\">(</span>string<span class=\\"token punctuation\\">:</span> <span class=\\"token string\\">\\"mapbox://styles/mapbox/streets-v10\\"</span><span class=\\"token punctuation\\">)</span>",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 55.349999999999994,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<span class=\\"token keyword\\">let</span> mapView <span class=\\"token operator\\">=</span> <span class=\\"token function\\">MGLMapView</span><span class=\\"token punctuation\\">(</span>frame<span class=\\"token punctuation\\">:</span> view<span class=\\"token punctuation\\">.</span>bounds<span class=\\"token punctuation\\">,</span> styleURL<span class=\\"token punctuation\\">:</span> url<span class=\\"token punctuation\\">)</span>",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 55.349999999999994,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "mapView<span class=\\"token punctuation\\">.</span>autoresizingMask <span class=\\"token operator\\">=</span> <span class=\\"token punctuation\\">[</span><span class=\\"token punctuation\\">.</span>flexibleWidth<span class=\\"token punctuation\\">,</span> <span class=\\"token punctuation\\">.</span>flexibleHeight<span class=\\"token punctuation\\">]</span>",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 55.349999999999994,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "mapView<span class=\\"token punctuation\\">.</span><span class=\\"token function\\">setCenter</span><span class=\\"token punctuation\\">(</span><span class=\\"token function\\">CLLocationCoordinate2D</span><span class=\\"token punctuation\\">(</span>latitude<span class=\\"token punctuation\\">:</span> <span class=\\"token number\\">59.31</span><span class=\\"token punctuation\\">,</span> longitude<span class=\\"token punctuation\\">:</span> <span class=\\"token number\\">18.06</span><span class=\\"token punctuation\\">)</span><span class=\\"token punctuation\\">,</span> zoomLevel<span class=\\"token punctuation\\">:</span> <span class=\\"token number\\">9</span><span class=\\"token punctuation\\">,</span> animated<span class=\\"token punctuation\\">:</span> <span class=\\"token boolean\\">false</span><span class=\\"token punctuation\\">)</span>",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 55.349999999999994,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "view<span class=\\"token punctuation\\">.</span><span class=\\"token function\\">addSubview</span><span class=\\"token punctuation\\">(</span>mapView<span class=\\"token punctuation\\">)</span>",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
            <div
              className="pr12"
              style={
                Object {
                  "paddingLeft": 40.9,
                }
              }
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<span class=\\"token punctuation\\">}</span>",
                  }
                }
                style={
                  Object {
                    "marginLeft": 14.45,
                    "textIndent": -14.45,
                  }
                }
              />
            </div>
          </div>
        </code>
      </pre>
      <div
        className="absolute z2 top right mr6 mt6 color-white"
      />
    </div>
  </div>
</div>
`;
